### DISCUSSION TITLE
Start up script order

### DISCUSSION THREAD

**Discussion by gpechiro** [coder/coder#11264] [Feature Request]
Is there a way to set the order that the coder_scripts with run_on_start=true run in? We have our own way to clone repos. I would like to make this into a company module so that it can easily be reused. There is also workspace specific tasks I would like to do like starting the dependencies the project needs. This can require changing directories into the repo that was cloned and starting docker containers using the project's docker compose file. 

The issue I am running into is the scripts all seem to be running asynchronously. While I understand this is an advantage for overall start up time it is causing issues when one script tries to change directories before the repo has finished cloning.

Right now I have to put all all of the scripts together in the coder_agent startup_script to ensure they run in the order I need. This works but it means I have duplicate non sharable code. I have tried using the [depends_on](https://developer.hashicorp.com/terraform/language/meta-arguments/depends_on) for terraform resources but the script still run asynchronously.

**Comment by matifali** [comment]
Hi @gpechiro, this is currently not possible. However, there is an existing feature request to support this behavior. Please see this issue for more context.  https://github.com/coder/coder/issues/10352
